libdnf/sack/query.cpp

```diff
@@ -1915,11 +1925,11 @@ Query::Impl::filterAdvisory(const Filter & f, Map *m, int keyname)
             }
 
             Id id = -1;
-            // Add to candidates resultPset pkgs that match name and arch with some already installed pkg
+            // Add to candidates resultPset pkgs that match name and arch with some already installed pkg or match name and either the installed or candidate are NOARCH
             while ((id = resultPset->next(id)) != -1) {
                 Solvable * s = pool_id2solvable(pool, id);
-                auto low = std::lower_bound(installed_solvables.begin(), installed_solvables.end(), s, NameArchSolvableComparator);
-                if (low != installed_solvables.end() && s->name == (*low)->name && s->arch == (*low)->arch) {
+                auto low = std::lower_bound(installed_solvables.begin(), installed_solvables.end(), s, NameSolvableComparator);
+                if (low != installed_solvables.end() && s->name == (*low)->name && (s->arch == (*low)->arch || s->arch == ARCH_NOARCH || (*low)->arch == ARCH_NOARCH)) {
```

Comment:
I am not sure whether it will work correctly for multiarch when one is installed.
I have installed `pkgA.i686` and `pkg.x86_64`. Then I have the candidate `pkgA.i686`, but the first mach in installed_solvable will be `pkg.x86_64`. Then the candidate will be not valid and not pushed into candidates. I recommend to test all installed_solvable with the same name. What do you think?

Reply:
I have added a loop over all the `installed_solvable` with matching `name`.

Reply:
LGTM